sdk/src/main/java/tech/relaycorp/sdk/Client.kt

```diff
@@ -0,0 +1,29 @@
+package tech.relaycorp.sdk
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+
+internal class Client {
+
+    // Gateway
+
+    suspend fun bind() {}
+    suspend fun unbind() {}
+
+    // First-Party Endpoints
+
+    suspend fun registerEndpoint(): FirstPartyEndpoint = FirstPartyEndpoint("")
+    suspend fun removeEndpoint(endpoint: FirstPartyEndpoint) {}
+    suspend fun listEndpoints(): List<FirstPartyEndpoint> = emptyList()
```

Comment:
I already used it in Echo, it's useful to figure out if we've done the endpoint configuration or not for Echo.

Reply:
Can you think of a reason why a real-world app would need that? I think this is one of those few things where the echo app is an outlier: If the echo app were not a throwaway prototype, we would be storing the addresses/ids of those endpoints, instead of assuming that the order is reliable enough to tell which is which:

```kotlin
            val endpoints = RelaynetClient.listEndpoints()
            val senderEndpoint = endpoints.first()
            val receiverEndpoint = endpoints.last()
```

In a production app, we should be doing something like the following instead:

```kotlin
            val senderEndpoint = FirstPartyEndpoint.retrieve(preferences.getSenderAddress())
            val receiverEndpoint = FirstPartyEndpoint.retrieve(preferences.getRecipientAddress())
```

More generally, I expect every real-world app to manage the mapping of business entities to Relaynet endpoints in its own DB or config file. For example, consider the following services we know about:

- Letro: This app will have a DB table for the contacts, which must contain Relaynet-related columns like `sender_endpoint_address` and `recipient_endpoint_address`, addition to things like `full_name` and `profile_picture`.
- Ping: This app will have to have its own DB table of third party endpoints, where `address` and `first_party_endpoint_address` will be some of the columns, along with any business-related columns we'd like to attach -- Like `creation_date` or `alias`, for example. (Or, if not a DB table, a config file.)

I know we'll need the `listEndpoints()` method internally, but unless we can anticipate a reason why a real-world app would need it, I'd rather keep it `internal` for now in order to keep the public API simpler and, more importantly, dig into why someone would need it public (if someone asks for it). My feeling is that asking for this to be public will either be a code smell or symptom that we lack an important feature.